### ISSUE TITLE
Bug: sort-switch-case AssertionError [ERR_ASSERTION]: Fix objects must not be overlapped in a report.

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [azat-io/eslint-plugin-perfectionist#335]
### Describe the bug

```plaintext
AssertionError [ERR_ASSERTION]: Fix objects must not be overlapped in a report.
    at mergeFixes (/Users/josh/repos/ts-api-utils/node_modules/.pnpm/eslint@9.13.0_jiti@2.3.3/node_modules/eslint/lib/linter/report-translator.js:165:9)
    at normalizeFixes (/Users/josh/repos/ts-api-utils/node_modules/.pnpm/eslint@9.13.0_jiti@2.3.3/node_modules/eslint/lib/linter/report-translator.js:197:16)
    at /Users/josh/repos/ts-api-utils/node_modules/.pnpm/eslint@9.13.0_jiti@2.3.3/node_modules/eslint/lib/linter/report-translator.js:371:49
    at FileContext.report (/Users/josh/repos/ts-api-utils/node_modules/.pnpm/eslint@9.13.0_jiti@2.3.3/node_modules/eslint/lib/linter/linter.js:1048:41)
    at file:///Users/josh/repos/ts-api-utils/node_modules/.pnpm/eslint-plugin-perfectionist@3.9.1_eslint@9.13.0_jiti@2.3.3__typescript@5.6.3/node_modules/eslint-plugin-perfectionist/dist/rules/sort-switch-case.mjs:146:21
    at pairwise (file:///Users/josh/repos/ts-api-utils/node_modules/.pnpm/eslint-plugin-perfectionist@3.9.1_eslint@9.13.0_jiti@2.3.3__typescript@5.6.3/node_modules/eslint-plugin-perfectionist/dist/utils/pairwise.mjs:7:9)
    at SwitchStatement (file:///Users/josh/repos/ts-api-utils/node_modules/.pnpm/eslint-plugin-perfectionist@3.9.1_eslint@9.13.0_jiti@2.3.3__typescript@5.6.3/node_modules/eslint-plugin-perfectionist/dist/rules/sort-switch-case.mjs:109:9)
    at ruleErrorHandler (/Users/josh/repos/ts-api-utils/node_modules/.pnpm/eslint@9.13.0_jiti@2.3.3/node_modules/eslint/lib/linter/linter.js:1084:48)
    at /Users/josh/repos/ts-api-utils/node_modules/.pnpm/eslint@9.13.0_jiti@2.3.3/node_modules/eslint/lib/linter/safe-emitter.js:45:58
    at Array.forEach (<anonymous>)
```

### Code example

https://github.com/JoshuaKGoldberg/ts-api-utils/blob/9ee03183986e93e573fa11ed7643b440603439a9/src/usage/getUsageDomain.ts

### ESLint version

9.13.0

### ESLint Plugin Perfectionist version

3.9.1

### Additional comments

Related to: https://github.com/eslint/eslint/issues/19040

Apologies I don't have a more isolated reproduction, I ran out of time trying to set one up. It's tricky. 🙂

### Validations

- [X] Read the [docs](https://perfectionist.dev).
- [X] Check that there isn't [already an issue](https://github.com/azat-io/eslint-plugin-perfectionist/issues) that reports the same bug to avoid creating a duplicate.

**Comment by hugop95** [comment]
Hi @JoshuaKGoldberg, the issue is related to the inline comment line 95. While waiting for a fix, you may remove that comment temporarily to re-enable ESLint to work again in that file.

**Comment by azat-io** [comment]
Added in 140c2c858ce6bed5ea0bac57f2cc3e4d9b6897fe.
Released in [v4.0.0](https://github.com/azat-io/eslint-plugin-perfectionist/releases/tag/v4.0.0).

If you'd like, you can support the release with a retweet:
https://x.com/azat_io_en/status/1858853714238271530
